chain/chain/src/chain.rs

```diff
@@ -699,6 +699,16 @@ impl Chain {
         };
         store_update.commit()?;
 
+        // We must load in-memory tries here, and not inside runtime, because
+        // if we were initializing from genesis, the runtime would be
+        // initialized when no blocks or flat storage were initialized. We
+        // require flat storage in order to load in-memory tries.
+        // TODO(#9511): The calculation of shard UIDs is not precise in the case
+        // of resharding. We need to revisit this.
+        let tip = store.head()?;
+        let shard_uids = epoch_manager.get_shard_layout(&tip.epoch_id)?.get_shard_uids();
+        runtime_adapter.load_mem_tries_on_startup(&shard_uids)?;
```

Comment:
Let's create a separate issue for this because it increases amount of steps to initialize NightshadeRuntime properly. I just remembered that at this point we don't even have FlatStorage object - it is created in Client::new -> FlatStorageCreator::new!
As it was needed only for migration, maybe we should do the following:
* in the basic case, move initialization of flat storage objects and tries into NightshadeRuntime
* in genesis case, repurpose method set_flat_storage_for_genesis to init both FS and tries from freshly created genesis.

Not insisting to do it now, but I'd like to fix it soon, because it's a big point for onboarding into runtime
cc @shreyan-gupta @pugachAG